R/apply_row_grp_plan.R

```diff
@@ -0,0 +1,108 @@
+#' Apply row group block formatting to data
+#'
+#' @param .data data
+#' @param row_grp_plan row group plan object
+#' @param group symbolic list of grouping
+#'
+#' @noRd
+#' @importFrom dplyr tibble mutate group_by arrange slice group_map case_when left_join row_number select
+#' @importFrom purrr map map2_dfr
+#' @importFrom tidyr unnest nest
+#' @importFrom rlang !!!
+apply_row_grp_plan <- function(.data, row_grp_plan, group, ...){
+
+  # determine which rows each block applies to
+  .data <- .data %>%
+    mutate(TEMP_row = row_number())
```

Comment:
is the assumption that the order in which the data are fed in is the order in which it should come out?

Reply:
@thebioengineer That was my assumption, since this gets applied after the pivot_wider and arrange based on sorting cols.